### DISCUSSION TITLE
About build version

### DISCUSSION THREAD

**Discussion by g-bougard** [glpi-project/glpi-agentmonitor#4] [Q&A]
Hi @redddcyclone,

I've a simple question. What's the best process to change the version of the exe ?
Is this just editing the `VS_VERSION_INFO VERSIONINFO` section from the `GLPI-AgentMontior.rc` file before the build or is there a simpler way to change it on the fly like providing a define option to msbuild ?
My concern is just to know if I can set it during the build workflow so I can eventually update it before the build. For example, if I tag a commit with `1.0.1` to build a new version to be signed with our future not expired SSL code-signing certificate, I would to just trigger a new build without needing to change the code.

For now, my idea is to create a `version.h` which should be included by `GLPI-AgentMontior.rc`. I'll try this way in a dedicated branch.

**Comment by redddcyclone** [comment]
Hi @g-bougard 

For this to be done correctly, the `VS_VERSION_INFO` block should be moved to a secondary `GLPI-AgentMonitor.rc2` file included in the `.rc` file via a `TEXTINCLUDE` block, because Visual Studio re-generates the `.rc` file every time it is edited (which may occur rather frequently as graphical elements such as the main window, icons and such are defined in the resource file), and it replaces the manually set constants from `version.h` with fixed values when doing so. 

I just committed an example of this to my branch [feat/external-version](https://github.com/redddcyclone/glpi-agentmonitor/tree/feat/external-version). Also, I included the files in the `.vcxproj` file. Check it out.

**Reply by g-bougard** [reply]
Sorry, I didn't see you comment before I made the PR I worked on this morning.

**Reply by g-bougard** [reply]
I updated my PR to follow your advice.

**Reply by redddcyclone** [reply]
It's fine, I didn't see you were already working on it 😅 I'm gonna check it out.
